mongodbatlas/fw_data_source_mongodbatlas_database_user.go

```diff
@@ -0,0 +1,217 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas/description"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
```

Comment:
since the resource is being rewritten any reason why not move to the new SDK?

Reply:
Thanks for flagging this.  The only reason that I can think of is that migrating to the new SDK is not a goal of the scope doc as it may add additional risk to the migration. However, I do agree that it is convenient to do it as part of this PR. @maastha @AgustinBettati Do you agree?

Reply:
I don't think I'd advise that, at least not for Stage 1, the idea of doing the migration in phases is for us to ensure there are no unexpected changes/behavior introduced, hence we should keep migration changes isolated for now.

Reply:
agree with @maastha

Reply:
Thanks for the discussion